sources/hyper-proxy-shim/src/stream.rs

```diff
@@ -0,0 +1,91 @@
+use std::io;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use tokio::io::{AsyncRead, AsyncWrite, ReadBuf};
```

Comment:
Need to figure out what license header to put in these files.

I'd expect something along the lines of what we do for the [cargo macros](https://github.com/bottlerocket-os/bottlerocket-sdk/blob/6d7ae094de951a92ea564a3c7ccd164a858d19b4/macros/cargo#L4).